src/backend/joanie/tests/core/test_utils_issuers.py

```diff
@@ -0,0 +1,79 @@
+"""Tests issuers in utils"""
+from io import BytesIO
+
+from django.template.exceptions import TemplateDoesNotExist
+from django.test import TestCase
+
+from pdfminer.high_level import extract_text as pdf_extract_text
+
+from joanie.core import enums, factories
+from joanie.core.utils import issuers
+
+
+class UtilsDocumentCreateTestCase(TestCase):
+    """Testcase for utility method to generate document"""
+
+    def test_utils_issuer_generate_document(self):
+        """
+        Generate document should create the pdf document
+        when the given path to reach the .html and .css file
+        exists.
+        """
+        user = factories.UserFactory()
+        factories.AddressFactory.create(owner=user)
+        order = factories.OrderFactory(
+            owner=user,
+            product=factories.ProductFactory(type=enums.PRODUCT_TYPE_CERTIFICATE),
+        )
+        contract = factories.ContractFactory(
+            order__owner=user,
+        )
+        markdown_content = """
+        ## Article 1
+        The student must have a computer to follow the course
+
+        ## Article 2
+        The student and the organization are tied down to this contract
+
+        ## Article 3
+        The student has paid in advance the whole course before the start
+        """
+        contract.definition.body = markdown_content
+        document_text = pdf_extract_text(
+            BytesIO(
+                issuers.generate_document(
+                    path="contract_definition",
+                    context=contract.definition.get_document_context(order),
```

Comment:
you are testing the generate_document util, not the contract definition so you should directly pass a context without any use of a contract definition.